neuland/simulation/generator/generator.h

```diff
@@ -0,0 +1,137 @@
+#pragma once
+#include <Math/GenVector/Cartesian3D.h>
+#include <Math/GenVector/Polar3D.h>
+#include <Math/GenVector/PxPyPzE4D.h>
+#include <algorithm>
+#include <cmath>
+#include <fmt/format.h>
+#include <TRandom.h>
+
+
+template <typename ADist, typename EDist>
+class TrackGenerator
+{
+  public:
+    TrackGenerator(const ADist& AngleDist,const EDist& EnergyDist)
+        : AngleDist_{ AngleDist }
+        , EnergyDist_{ EnergyDist }
+    {
+    }
+    void set_detector_width(double width);
+    void set_detector_length(double length);
+    void set_detector_height(double height);
+
+  private:
+    double detector_width{};
+    double detector_length{};
+    double detector_height{};
+
+    double particle_mass{};
+    static constexpr double speed_of_light{ 299792458 };
+
+    ROOT::Math::Cartesian3D<double> position{};
+
+    const ADist& AngleDist_;
+    const EDist& EnergyDist_;
+    TRandom rd_engine_;
```

Comment:
You shouldn't define your own random generator, but rather use the random generator available in the whole program. Therefore, here it's should be a pointer set by users. If users forget to set it, use the default random generator called `gRandom` (a global variable defined by ROOT).